### PR TITLE
Develop upstream sync 221219

### PR DESCRIPTION
**Pull request by i-chaochen**
2 disable unit tests

1. xla_ops_test  https://github.com/ROCmSoftwarePlatform/frameworks-internal/issues/3221
 
2. nn_test_gpu   https://github.com/ROCmSoftwarePlatform/frameworks-internal/issues/3222

Not sure the first one is because GemmRewriter is disable, but we definitely shall add our's soon.

Also, I think we can improve the change for the choice between se::RocmComputeCapability and se::CudaComputeCapability by GpuVersion, as it's used to abstract Gpu Hardware version.
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/gpu/gpu_types.h#L29-L37
```
using GpuVersion =
    std::variant<se::CudaComputeCapability, se::RocmComputeCapability>;
}
```

This improvement also can go to upstream as well.

### CODE REVIEW

**Comment by i-chaochen** [comment]
retest Ubuntu-GPU-single please

**Comment by i-chaochen** [comment]
retest Ubuntu-sanity please

**Comment by i-chaochen** [comment]
retest Ubuntu-GPU-single please

**Comment by i-chaochen** [comment]
retest Ubuntu-GPU-multi please

**Comment by i-chaochen** [comment]
retest Ubuntu-sanity please

**Comment by jayfurmanek** [comment]
retest Ubuntu-CPU please

**Comment by jayfurmanek** [comment]
restest Ubuntu-sanity please

File: tensorflow/compiler/xla/service/gpu/tree_reduction_rewriter.cc [hunk 1]
```diff
@@ -256,16 +261,23 @@ class ReductionRewriterVisitor : public DfsHloRewriteVisitor {
...
 };
 
 StatusOr<bool> GpuTreeReductionRewriter::Run(
     HloModule *module,
     const absl::flat_hash_set<absl::string_view> &execution_threads) {
   VLOG(5) << "Rewriter input: " << module->ToString();
   TF_ASSIGN_OR_RETURN(bool changed,
+#if GOOGLE_CUDA
                       ReductionRewriterVisitor(cuda_compute_capability_)
+#elif TENSORFLOW_USE_ROCM
```
Comment (jayfurmanek) [inline tensorflow/compiler/xla/service/gpu/tree_reduction_rewriter.cc]: This is causing the issue with the CPU test. When CUDA or ROCM are not defined we end up with a dangling RunOnModule()

Comment (i-chaochen) [reply tensorflow/compiler/xla/service/gpu/tree_reduction_rewriter.cc]: Fixed it. Please check.

**Review by jayfurmanek — APPROVED** [review]

**Pull request merged**